apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/KotlinResolver.kt

```diff
@@ -156,6 +159,43 @@ class KotlinResolver(entries: List<ResolverEntry>, val next: KotlinResolver?) {
     }
   }
 
+  private fun nonNullableScalarAdapterInitializer(type: IrScalarType): CodeBlock {
+    return when (val adapterInitializer = scalarMapping[type.name]?.adapterInitializer) {
+      is ExpressionAdapterInitializer -> {
+        CodeBlock.of(adapterInitializer.expression)
+      }
+      is RuntimeAdapterInitializer -> {
+        val target = resolve(ResolverKeyKind.CustomScalarTarget, type.name)
+        CodeBlock.of(
+            "$customScalarAdapters.responseAdapterFor<%T>(%L)",
+            target,
+            resolveCompiledType(type.name)
+        )
+      }
+      else -> {
+        when (type.name) {
+          "Boolean" -> CodeBlock.of("%M", KotlinSymbols.BooleanAdapter)
+          "ID" -> CodeBlock.of("%M", KotlinSymbols.StringAdapter)
+          "String" -> CodeBlock.of("%M", KotlinSymbols.StringAdapter)
+          "Int" -> CodeBlock.of("%M", KotlinSymbols.IntAdapter)
+          "Float" -> CodeBlock.of("%M", KotlinSymbols.DoubleAdapter)
```

Comment:
I'd love this to be moved up a few layers so that the resolver can apply the same logic everywhere. Not really sure how to do this yet but I might investigate this in a follow up PR unless you have ideas how to do this already.